pbj-integration-tests/src/test/java/com/hedera/pbj/intergration/test/HashEqualsTest.java

```diff
@@ -0,0 +1,92 @@
+package com.hedera.pbj.intergration.test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+import org.junit.jupiter.api.Test;
+
+import com.hedera.pbj.test.proto.pbj.TimestampTest;
+import com.hedera.pbj.test.proto.pbj.TimestampTest2;
+
+class HashEqualsTest {
+    @Test
+    void differentObjectsWithDefaulEquals() {
+        TimestampTest tst = new TimestampTest(1, 2);
+        TimestampTest2 tst2 = new TimestampTest2(1, 2, 0);
+
+        assertFalse(tst.equals(tst2));
```

Comment:
0 is the default value for int32, why are these two objects considered not equal?

Reply:
It's a repeated field. If the number of entries is different, the object is not equal.